src/cxp/controller.ts

```diff
@@ -80,7 +80,8 @@ function environmentFilter<S extends ConfigurationSubject, CC extends Configurat
  */
 export function createController<S extends ConfigurationSubject, C extends Settings>(
     context: Context<S, C>,
-    createMessageTransports: (extension: ConfiguredExtension, options: ClientOptions) => Promise<MessageTransports>
+    createMessageTransports: (extension: ConfiguredExtension, options: ClientOptions) => Promise<MessageTransports>,
+    experimentalClientCapabilities?: any
```

Comment:
It seems to me that experimentalClientCapabilities are a part of the `context: Context<S, C>`, which is documented as:

> Description of the context in which extensions-client-common is running, and platform-specific hooks.

The experimentalClientCapabilities fit into that description. So I would recommend adding a `experimentalClientCapabilities: any` field to that `Context` type.

(Another way to look at it is that ~50% of the other fields on `Context` could be passed in as an arg to `createController`, but we chose to centralize them in `Context`. I don't see an argument for why `experimentalClientCapabilities` is different in a relevant way from `updateConfiguration`, `configurationCascade`, etc. The nice thing about putting these in `Context` is that ALL of the platform-specific stuff lives in one place, instead of needing to customize an argument here, a field here, etc.)

Reply:
Ah good call, I had overlooked `Context`